File: support/mock_backend/devLib/gertboard.c
#include <stdint.h>
#include <stddef.h>
#include <stdbool.h>
#include <string.h>
#include "devLib/gertboard.h"
void gertboardAnalogWrite(const int chan, const int value) { (void)chan;(void)value; }
int gertboardAnalogRead(const int chan) { (void)chan;return 1; }
int gertboardSPISetup(void) { return 1; }
int gertboardAnalogSetup(const int pinBase) { (void)pinBase;return 1; }

File: tests/devLib/gertboard.py
import importlib
import pytest

m=importlib.import_module("wiringop.dev_lib.gertboard")

def test_gertboard_analog_write():
    _=m.gertboard_analog_write(1, 1)

def test_gertboard_analog_read():
    _=m.gertboard_analog_read(1)

def test_gertboard_spi_setup():
    _=m.gertboard_spi_setup()

def test_gertboard_analog_setup():
    _=m.gertboard_analog_setup(1)
